### PR TITLE
sexp: Add missing API: items/pairs/treeRepr/getKey

### PR DESCRIPTION
**Pull request by haxscramper**
- expose `.car` and `.cdr` fields for direct access - everything else can
  already be accesssed, but those were wrapped in weird `getCons()` proc
  that retunred both elements.
- `treeRepr` for S-expression printing
- `getKey` for S-expression keyword pair
- missing `pairs()` and `mpairs()`

### CODE REVIEW

File: nimsuggest/sexp.nim [hunk 1]
```diff
@@ -620,13 +626,73 @@ iterator items*(node: SexpNode): SexpNode =
...
 
+iterator mpairs*(node: var SexpNode): (int, var SexpNode) =
+  ## Iterator for the pairs of `node`. `node` has to be a SList. Items can be
+  ## modified.
+  assert node.kind == SList
+  for i, node in mpairs(node.elems):
+    yield (i, node)
+
+proc treeRepr*(node: SexpNode): string =
+  proc aux(node: SexpNode, level: int, res: var string) =
```
Comment (saem) [inline nimsuggest/sexp.nim:652]: Document the public procs please.

**Comment by saem** [comment]
I'd put it into the standard library as a next step, honestly super useful elsewhere and the nimsuggest based include is annoying. Could be a next step.

**Comment by haxscramper** [comment]
For now this is just a simple API update, in the future this can be moved to the stdlib, yes, I see no issues with that.

**Comment by saem** [comment]
Bors r+

**Pull request merged**